File: frontend/src/backend/noteService.js
import {config, appFetch} from './appFetch';

export const findNotesByPlayer = (playerId, onSuccess, onErrors) =>
    appFetch(`/notes?playerId=${playerId}`, config('GET'), onSuccess, onErrors);

export const findNoteById = (noteId, onSuccess, onErrors) =>
    appFetch(`/notes/${noteId}`, config('GET'), onSuccess, onErrors);

export const findNotesByPlayerAndDates = (playerId, startDate, endDate, onSuccess, onErrors) =>
    appFetch(`/notes/dates?playerId=${playerId}&startDate=${startDate}&endDate=${endDate}`, config('GET'), onSuccess, onErrors);

export const addNoteToPlayer = (playerId, title, description, onSuccess, onErrors) =>
    appFetch(`/notes/${playerId}/addNoteToPlayer?title=${title}&description=${description}`, config('POST'), onSuccess, onErrors);

export const updateNote = (noteId, title, description, onSuccess, onErrors) =>
    appFetch(`/notes/${noteId}?title=${title}&description=${description}`, config('PUT'), onSuccess, onErrors);

export const removeNote = (noteId, onSuccess, onErrors) =>
    appFetch(`/notes/${noteId}`, config('DELETE'), onSuccess, onErrors);

File: backend/src/main/java/es/udc/paproject/backend/model/services/NoteService.java
package es.udc.paproject.backend.model.services;

import java.time.LocalDateTime;
import java.util.List;

import es.udc.paproject.backend.model.entities.Note;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.StartDateAfterEndDateException;

public interface NoteService {
    
    Note addNoteToPlayer(Long playerId, String title, String description) throws InstanceNotFoundException;

    List<Note> findNotesByPlayer(Long playerId) throws InstanceNotFoundException;

    Note findNoteById(Long noteId) throws InstanceNotFoundException;

    List<Note> findNotesByPlayerAndDates(Long playerId, LocalDateTime startDate, LocalDateTime endDate) throws InstanceNotFoundException, StartDateAfterEndDateException;

    void removeNote(Long noteId) throws InstanceNotFoundException;

    Note updateNote(Long noteId, String title, String description) throws InstanceNotFoundException;
}